### PR TITLE
refactor: change makeDepositFacet to getDepositFacet

### LINKED ISSUE #1372: [ERTP] Pre-make deposit facet
```
* @property {() => DepositFacet} makeDepositFacet
```
Freshness doesn't do anything useful here. Prefer `getDepositFacet` and pre-make one and return the same one each time.

_Originally posted by @erights in https://github.com/Agoric/agoric-sdk/pull/1357_

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
fixes: #1372

### CODE REVIEW

File: packages/ERTP/src/issuer.js [hunk 1]
```diff
@@ -106,8 +106,12 @@ function makeIssuerKit(allegedName, amountMathKind = MathKind.NAT) {
       },
       getCurrentAmount: () => purseLedger.get(purse),
       getAllegedBrand: () => brand,
-      makeDepositFacet: () => harden({ receive: purse.deposit }),
+      // eslint-disable-next-line no-use-before-define
+      getDepositFacet: () => depositFacet,
     });
+
+    const depositFacet = harden({ receive: purse.deposit });
```
Comment (katelynsills) [inline packages/ERTP/src/issuer.js:113]: why not move this up so we don't have to use the eslint-disable?

**Review by katelynsills — APPROVED** [review]
Looks good! Let's update the documentation and dapps simultaneously with this change, so let's not merge this until we've gotten PRs for the dapps too. I've been recently reminded about how much nicer this method works, currently catching up on my own PRs.

File: packages/ERTP/src/issuer.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/ERTP/src/issuer.js:113]: If it's before purse, then purse needs the eslint-disable. If it's inside purse, then it's visible to purse holders. I could have used let, but that still needs an assignment after the purse is defined.

This seemed better than any of those possibilities. Did I miss another approach?

Comment (katelynsills) [reply packages/ERTP/src/issuer.js:113]: Oh, gotcha. Sorry, missed the circular reference. This looks fine

**Comment by Chris-Hibbert** [comment]
assocated doc changes in https://github.com/Agoric/documentation/pull/261

**Comment by Chris-Hibbert** [comment]
I didn't find any instances of `makeDepositFacet` in the dapp repositories.

**Comment by katelynsills** [comment]
> I didn't find any instances of `makeDepositFacet` in the dapp repositories.

That is a very good point. I realized the `receive` method didn't change, so there is nothing to change in the dapps, and actually, now nothing uses this depositFacet since the wallet uses Michael's separate implementation of a depositFacet. So there's two reasons the dapps wouldn't change because of this. D'oh.

**Comment by Chris-Hibbert** [comment]
> nothing uses this depositFacet since the wallet uses Michael's separate implementation of a depositFacet. 

Is that the observablePurse I changed in `.../dapp-svelte-wallet/api/src/observable.js` in this PR? Nothing seemed to call that either.

**Comment by katelynsills** [comment]
> > nothing uses this depositFacet since the wallet uses Michael's separate implementation of a depositFacet.
> > Is that the observablePurse I changed in `.../dapp-svelte-wallet/api/src/observable.js` in this PR? Nothing seemed to call that either.

I'm not sure exactly how it works, but I was thinking of this: https://github.com/Agoric/agoric-sdk/blob/a21832a9a2b88aa43d2b532a5b92f99c47d3e11b/packages/dapp-svelte-wallet/api/src/lib-wallet.js#L1010

**Comment by Chris-Hibbert** [comment]
> but I was thinking of this:

Okay, whatever that's doing doesn't seem to implicate receive on a depositFacet, or need to be updated here, so I'm going to go ahead and submit these two shortly if I don't hear an objection.

**Comment by katelynsills** [comment]
> > but I was thinking of this:
> 
> Okay, whatever that's doing doesn't seem to implicate receive on a depositFacet, or need to be updated here, so I'm going to go ahead and submit these two shortly if I don't hear an objection.

yup, it's a completely different implementation. We're good to go here 👍

**Pull request merged**